ee/hogai/trends/nodes.py

```diff
@@ -0,0 +1,398 @@
+import itertools
+import json
+import xml.etree.ElementTree as ET
+from functools import cached_property
+from typing import Union, cast
+
+from langchain.agents.format_scratchpad import format_log_to_str
+from langchain.agents.output_parsers import ReActJsonSingleInputOutputParser
+from langchain_core.agents import AgentAction, AgentFinish
+from langchain_core.exceptions import OutputParserException
+from langchain_core.messages import AIMessage, BaseMessage, merge_message_runs
+from langchain_core.output_parsers import PydanticOutputParser
+from langchain_core.prompts import ChatPromptTemplate, HumanMessagePromptTemplate
+from langchain_core.runnables import RunnableConfig, RunnableLambda
+from langchain_openai import ChatOpenAI
+from pydantic import ValidationError
+
+from ee.hogai.hardcoded_definitions import hardcoded_prop_defs
+from ee.hogai.trends.prompts import (
+    react_definitions_prompt,
+    react_follow_up_prompt,
+    react_scratchpad_prompt,
+    react_system_prompt,
+    react_user_prompt,
+    trends_group_mapping_prompt,
+    trends_new_plan_prompt,
+    trends_plan_prompt,
+    trends_question_prompt,
+    trends_system_prompt,
+)
+from ee.hogai.trends.toolkit import (
+    GenerateTrendTool,
+    TrendsAgentToolkit,
+    TrendsAgentToolModel,
+)
+from ee.hogai.trends.utils import GenerateTrendOutputModel
+from ee.hogai.utils import (
+    AssistantMessage,
+    AssistantNode,
+    AssistantNodeName,
+    AssistantState,
+    remove_line_breaks,
+)
+from posthog.hogql_queries.ai.team_taxonomy_query_runner import TeamTaxonomyQueryRunner
+from posthog.hogql_queries.query_runner import ExecutionMode
+from posthog.models.group_type_mapping import GroupTypeMapping
+from posthog.schema import (
+    CachedTeamTaxonomyQueryResponse,
+    TeamTaxonomyQuery,
+    VisualizationMessagePayload,
+)
+
+
+class CreateTrendsPlanNode(AssistantNode):
+    name = AssistantNodeName.CREATE_TRENDS_PLAN
+
+    @cached_property
+    def _events_prompt(self) -> str:
+        event_description_mapping = {
+            "$identify": "Identifies an anonymous user. This event doesn't show how many users you have but rather how many users used an account."
+        }
```

Comment:
This kinda seems like a poorer version of `hardcoded_prop_defs` – why not put this description directly in there?